packages/@react-spectrum/tree/docs/TreeView.mdx

```diff
@@ -0,0 +1,604 @@
+  {(item: MyItem) => (
+    <TreeViewItem childItems={item.childItems} textValue={item.name}>
+      <Text>{item.name}</Text>
+      {item.icon}
+      <ActionGroup onAction={(key) => alert(`Item: ${item.id}, Action: ${key}`)}>
+        <Item key="edit" textValue="Edit">
+          <Edit />
+          <Text>Edit</Text>
+        </Item>
+        <Item key="delete" textValue="Delete">
+          <Delete />
+          <Text>Delete</Text>
+        </Item>
+      </ActionGroup>
+    </TreeViewItem>
+  )}
+</TreeView>
+```
+
+### With action menus
+
+```tsx example
+type MyItem = {
+  id: string,
+  name: string,
+  icon: JSX.Element,
+  childItems?: MyItem[]
+};
+
+let items: MyItem[] = [
+  {id: 'projects', name: 'Projects', icon: <Folder />, childItems: [
+    {id: 'project-1', name: 'Project 1', icon: <FileTxt />},
+    {id: 'project-2', name: 'Project 2', icon: <Folder />, childItems: [
+      {id: 'document-a', name: 'Document A', icon: <FileTxt />},
+      {id: 'document-b', name: 'Document B', icon: <FileTxt />},
+    ]}
+  ]},
+  {id: 'reports', name: 'Reports', icon: <Folder />, childItems: [
+    {id: 'report-1', name: 'Reports 1', icon: <FileTxt />}
+  ]}
+];
+
+<TreeView aria-label="Example tree with dynamic content" height="size-3000" items={items}>
+  {(item: MyItem) => (
+    <TreeViewItem childItems={item.childItems} textValue={item.name}>
+      <Text>{item.name}</Text>
+      {item.icon}
+      <ActionMenu onAction={(key) => alert(`Item: ${item.id}, Action: ${key}`)}>
+        <Item key="edit" textValue="Edit">
+          <Edit />
+          <Text>Edit</Text>
+        </Item>
+        <Item key="delete" textValue="Delete">
+          <Delete />
+          <Text>Delete</Text>
+        </Item>
+      </ActionMenu>
+    </TreeViewItem>
+  )}
+</TreeView>
+```
+
+## Props
+
+### TreeView props
+
+<PropTable component={docs.exports.TreeView} links={docs.links} style={{marginBottom: '40px'}} />
+
+### TreeViewItem props
+
+<PropTable component={docs.exports.TreeViewItem} links={docs.links} style={{marginBottom: '40px'}} />
+
+## Visual options
+### Empty state
+Use the `renderEmptyState` prop to customize what the TreeView will display if there are no items provided.
+
+```tsx example
+import {Content} from '@react-spectrum/view';
+import {IllustratedMessage} from '@react-spectrum/illustratedmessage';
+import NotFound from '@spectrum-icons/illustrations/NotFound';
+import {Heading} from '@react-spectrum/text';
+
+function renderEmptyState() {
+  return (
+    <IllustratedMessage>
+      <NotFound />
+      <Heading>No results</Heading>
+      <Content>No results found</Content>
+    </IllustratedMessage>
+  );
+}
+
+<TreeView aria-label="Example tree for empty state" height="size-2400" renderEmptyState={renderEmptyState}>
+  {[]}
+</TreeView>
+```
+
+## Testing
+
+The TreeView features automatic virtualization and may need specific mocks in a test environment to enable said virtualization properly.
```

Comment:
The TreeView isn't actually virtualized so this sentence will need to be removed along with the link below. Thanks for including this section though :)